front/src/modules/powerRestriction/helpers/formatPowerRestrictionRangesWithHandled.ts

```diff
@@ -29,49 +32,42 @@ export const formatPowerRestrictionRanges = (
         return null;
       }
       return {
-        start: mmToM(stepsPathPositions[startStep]),
-        stop: mmToM(stepsPathPositions[stopStep]),
-        code: powerRestriction.value,
+        position: {
+          start: stepsPathPositions[startStep],
+          end: stepsPathPositions[stopStep],
+        },
+        value: { powerRestriction: powerRestriction.value },
       };
     })
   );
 
 /** Format power restrictions data to be used in simulation results charts */
-export const addHandledToPowerRestrictions = (
-  powerRestrictionRanges: Omit<SimulationPowerRestrictionRange, 'handled'>[],
+export const addHandledToPowerRestrictionsAndConvertPositions = (
+  powerRestrictionRanges: LayerData<Omit<PowerRestrictionValues, 'handled'>>[],
   voltageRanges: RangedValue[],
   rollingStockEffortCurves: RollingStock['effort_curves']['modes']
-): SimulationPowerRestrictionRange[] => {
+): LayerData<PowerRestrictionValues>[] => {
   const powerRestrictionsByMode = getRollingStockPowerRestrictionsByMode(rollingStockEffortCurves);
 
-  const restrictionsWithHandled: SimulationPowerRestrictionRange[] = [];
-
-  powerRestrictionRanges.forEach((powerRestrictionRange) => {
-    const powerRestrictionBeginInMm = mToMm(powerRestrictionRange.start);
-    const powerRestrictionEndInMm = mToMm(powerRestrictionRange.stop);
+  const restrictionsWithHandled: LayerData<PowerRestrictionValues>[] = [];
 
+  powerRestrictionRanges.forEach(({ position: { start, end }, value: { powerRestriction } }) => {
     // find all the voltage ranges which overlap the powerRestrictionRange
     voltageRanges.forEach((voltageRange) => {
```

Comment:
```suggestion
    voltageRanges.forEach(({ begin, end }) => {
    --- optional --- (to avoid using mToMm 2 times for restriction begin and end)
    const beginInMm = mToMm(begin);
    const endInMm = mToMm(end);
    ---
```

Reply:
good idea, but end is already declared for position line 54

Reply:
Ok so we could do (I missed also `value`)

```suggestion
    voltageRanges.forEach(({ begin, end: voltageRangeEnd, value }) => {
    --- optional --- (to avoid using mToMm 2 times for restriction begin and end)
    const beginInMm = mToMm(begin);
    const endInMm = mToMm(voltageRangeEnd);
```

Reply:
done